### PR TITLE
 Fix WPCS: WordPress, Squiz and Generic

### PR DESCRIPTION
**Pull request by pedro-mendonca**
See #585 

Rules fixed
- [x] `Squiz.PHP.EmbeddedPhp.SpacingAfterOpen`
- [x] `Squiz.PHP.EmbeddedPhp.ContentAfterOpen`
- [x] `Squiz.PHP.EmbeddedPhp.ContentBeforeEnd`
- [x] `Squiz.PHP.EmbeddedPhp.ContentBeforeOpen`
- [x] `Squiz.PHP.EmbeddedPhp.NoSemicolon`
- [x] `Squiz.PHP.EmbeddedPhp.SpacingBeforeClose`
- [x] `Squiz.WhiteSpace.SemicolonSpacing.Incorrect`
- [x] `Squiz.Commenting.BlockComment.NoNewLine`
- [x] `Squiz.Commenting.FunctionComment.SpacingAfterParamType`
- [x] `Squiz.Commenting.InlineComment.NoSpaceBefore`
- [x] `Squiz.Strings.DoubleQuoteUsage.NotRequired`
- [x] `Squiz.ControlStructures.ControlSignature.SpaceAfterCloseBrace`
- [x] `Squiz.Strings.ConcatenationSpacing`
- [x] `Squiz.Functions.FunctionDeclarationArgumentSpacing.SpaceAfterEquals`
- [x] `WordPress.WhiteSpace.ControlStructureSpacing.NoSpaceBetweenStructureColon`
- [x] `WordPress.Classes.ClassInstantiation.MissingParenthesis`
- [x] `WordPress.Arrays.MultipleStatementAlignment.LongIndexSpaceBeforeDoubleArrow`
- [x] `Generic.Formatting.MultipleStatementAlignment.NotSameWarning`

### CODE REVIEW

File: gp-includes/route.php [hunk 1]
```diff
@@ -311,10 +311,10 @@ public function set_notices_and_errors() {
 	/**
 	 * Loads a template.
 	 *
-	 * @param string $template template name to load
-	 * @param array $args Associative array with arguements, which will be exported in the template PHP file
+	 * @param string      $template template name to load
+	 * @param array       $args Associative array with arguements, which will be exported in the template PHP file
 	 * @param bool|string $honor_api If this is true or 'api' and the route is processing an API request
-	 * 		the template name will be suffixed with .api. The actual file loaded will be template.api.php
+	 *      the template name will be suffixed with .api. The actual file loaded will be template.api.php
```
Comment (ocean90) [inline gp-includes/route.php]: This should be aligned with where the description starts at "If"

File: gp-includes/router.php [hunk 2]
```diff
@@ -59,97 +59,97 @@ private function default_routes() {
...
+			"get:(/languages)/$locale/$dir/glossary"       => array( 'GP_Route_Glossary_Entry', 'glossary_entries_get' ),
+			"post:(/languages)/$locale/$dir/glossary"      => array( 'GP_Route_Glossary_Entry', 'glossary_entries_post' ),
+			"post:(/languages)/$locale/$dir/glossary/-new" => array( 'GP_Route_Glossary_Entry', 'glossary_entry_add_post' ),
 			"post:(/languages)/$locale/$dir/glossary/-delete" => array( 'GP_Route_Glossary_Entry', 'glossary_entry_delete_post' ),
-			"get:(/languages)/$locale/$dir/glossary/-export"  => array( 'GP_Route_Glossary_Entry', 'export_glossary_entries_get' ),
-			"get:(/languages)/$locale/$dir/glossary/-import"  => array( 'GP_Route_Glossary_Entry', 'import_glossary_entries_get' ),
+			"get:(/languages)/$locale/$dir/glossary/-export" => array( 'GP_Route_Glossary_Entry', 'export_glossary_entries_get' ),
+			"get:(/languages)/$locale/$dir/glossary/-import" => array( 'GP_Route_Glossary_Entry', 'import_glossary_entries_get' ),
 			"post:(/languages)/$locale/$dir/glossary/-import" => array( 'GP_Route_Glossary_Entry', 'import_glossary_entries_post' ),
```
Comment (ocean90) [inline gp-includes/router.php]: This looks odd, see https://github.com/pedro-mendonca/GlotPress-WP/blob/1f29dc7138c98cb185e57ae7f4243f5cac915e24/gp-includes/router.php#L77-L83

File: gp-includes/routes/project.php [hunk 3]
```diff
@@ -398,7 +395,7 @@ public function permissions_post( $project_path ) {
 			if ( $this->invalid_and_redirect( $new_permission, gp_url_current() ) ) {
 				return;
 			}
-			$permission = GP::$validator_permission->create( $new_permission );
+			$permission          = GP::$validator_permission->create( $new_permission );
```
Comment (ocean90) [inline gp-includes/routes/project.php]: This should stay as is I think since the next line is a condition.

File: gp-templates/project.php [hunk 4]
```diff
@@ -48,12 +49,16 @@ function( $items ) {
...
 <?php endif; ?>
 
-<div id="project" <?php if ( $sub_projects ) {
-	echo ' class="with-sub-projects"'; } ?>>
+<div id="project" 
+<?php
+if ( $sub_projects ) {
+	echo ' class="with-sub-projects"'; }
+	?>
+	>
```
Comment (ocean90) [inline gp-templates/project.php]: This will cause a lot of unnecessary whitespace in the HTML markup. The following is the preferred approach.

```php
<?php
$class = $sub_projects ? ' class="with-sub-projects"' : ''; 
?>
<div id="project"<?php echo $class; ?>>

File: gp-templates/project.php [hunk 5]
```diff
@@ -178,8 +184,10 @@ function( $items ) {
 	<dt>
 		<?php gp_link_project( $sub_project, esc_html( $sub_project->name ) ); ?>
 		<?php gp_link_project_edit( $sub_project, null, array( 'class' => 'bubble' ) ); ?>
-		<?php if ( $sub_project->active ) {
-			echo "<span class='active bubble'>" . __( 'Active', 'glotpress' ) . "</span>"; } ?>
+		<?php
+		if ( $sub_project->active ) {
+			echo "<span class='active bubble'>" . __( 'Active', 'glotpress' ) . '</span>'; }
```
Comment (ocean90) [inline gp-templates/project.php]:
```suggestion
			echo "<span class='active bubble'>" . __( 'Active', 'glotpress' ) . '</span>';
		}
```
The closing bracket should be on the next line.

File: gp-templates/translations.php [hunk 6]
```diff
@@ -217,12 +217,12 @@
...
-							'waiting'                                     => __( 'Waiting approval', 'glotpress' ),
-							'rejected'                                    => __( 'Rejected', 'glotpress' ),
-							'untranslated'                                => __( 'Without current translation', 'glotpress' ),
-							'either'                                      => __( 'Any', 'glotpress' ),
+							'current'      => __( 'Current only', 'glotpress' ),
+							'old'          => __( 'Approved, but obsoleted by another translation', 'glotpress' ),
+							'waiting'      => __( 'Waiting approval', 'glotpress' ),
+							'rejected'     => __( 'Rejected', 'glotpress' ),
+							'untranslated' => __( 'Without current translation', 'glotpress' ),
+							'either'       => __( 'Any', 'glotpress' ),
```
Comment (ocean90) [inline gp-templates/translations.php]: I'm assuming that this is because of https://github.com/WordPress/WordPress-Coding-Standards/wiki/Customizable-sniff-properties#array-alignment-maximum-column. What about increasing the number so it can be aligned with `current_or_waiting_or_fuzzy_or_untranslated`?

**Review by ocean90 — CHANGES_REQUESTED** [review]

File: gp-includes/route.php (hunk 1 above)
Comment (pedro-mendonca) [reply gp-includes/route.php]: Done, also aligned comments and added fullstops.

File: gp-includes/router.php (hunk 2 above)
Comment (pedro-mendonca) [reply gp-includes/router.php]: I wonder why phpcbf didn't format this one

Comment (pedro-mendonca) [reply gp-includes/router.php]: Even if I fix it, phpcbf undoes it.
I've changed the arrays to multiline, to see if it's a false positive somehow, it still changes the format to a bad alignemnt, of only one space.
I'll ignore it for now and align the whole array as it should.

File: gp-includes/routes/project.php (hunk 3 above)
Comment (pedro-mendonca) [reply gp-includes/routes/project.php]: Sure, I thinh phpcs is forcing this because of the ternary operator format.
Another false positive I guess

File: gp-templates/project.php (hunk 4 above)
Comment (pedro-mendonca) [reply gp-templates/project.php]: Fixed, thanks for the suggestion.

File: gp-templates/project.php (hunk 5 above)
Comment (pedro-mendonca) [reply gp-templates/project.php]: Of course :)
Also fixed the indent of the closing ?> tag
Mental note: so much escaping to do! :)

File: gp-templates/translations.php (hunk 6 above)
Comment (pedro-mendonca) [reply gp-templates/translations.php]: I agree, I'm fixing the alignment.
What ammount do you suggest set it to?

Comment (pedro-mendonca) [reply gp-templates/translations.php]: maxColumn 80 seems to be enough

File: gp-includes/router.php (hunk 2 above)
Comment (pedro-mendonca) [reply gp-includes/router.php]: The issue here is also Arrays.MultipleStatementAlignment maxColumn.
Changed it to 80 and it's ok now.

**Comment by pedro-mendonca** [comment]
@ocean90 I'll push here the change of `Arrays.MultipleStatementAlignment` maxColumn to 80, it solves the issues of the arrays mentioned above in your review.

File: gp-includes/routes/project.php (hunk 3 above)
Comment (pedro-mendonca) [reply gp-includes/routes/project.php]: This will keep being signaled by phpcs, I'll separate with a line to break the phpcs need for alignment.

File: gp-templates/translations.php [hunk 7]
```diff
@@ -309,8 +309,12 @@
 	</form>
 </div>
 
-<table id="translations" class="translations clear<?php if ( 'rtl' == $locale->text_direction ) {
-	echo ' translation-sets-rtl'; } ?>">
+<table id="translations" class="translations clear
+<?php
+if ( 'rtl' == $locale->text_direction ) {
+	echo ' translation-sets-rtl'; }
+	?>
```
Comment (ocean90) [inline gp-templates/translations.php]: This should also be changed to only echo a variable, see the sub-project example.

**Review by ocean90 — CHANGES_REQUESTED** [review]

File: gp-templates/translations.php (hunk 7 above)
Comment (pedro-mendonca) [reply gp-templates/translations.php]: As it's not the only class, I'll call the variable `$class_rtl`.
Also, I think we should start escaping things, I'll use `esc_attr()`.

**Review by ocean90 — APPROVED** [review]

**Comment by ocean90** [comment]
Thank you!
